adlfs/spec.py

```diff
@@ -925,109 +1016,22 @@ async def _ls(
         if invalidate_cache:
             self.dircache.clear()
```

Comment:
We are invalidating cache in one single place here, so we don't need to pass `invalidate_cache` down because `if smth not in self.dircache` will just work.